go-controller/pkg/ovn/ovn.go

```diff
@@ -348,7 +348,9 @@ func (oc *Controller) Run(wg *sync.WaitGroup, nodeName string) error {
 		}
 		oc.egressFirewallDNS.Run(egressFirewallDNSDefaultDuration)
 		oc.egressFirewallHandler = oc.WatchEgressFirewall()
-
+		if config.Gateway.Mode == config.GatewayModeShared {
+			oc.WatchEgressFirewallNodes()
```

Comment:
could you add a comment as to why it is only needed for shared gateway mode?

Reply:
Sure